src/toast/ops/scan_map.py

```diff
@@ -32,6 +33,8 @@ class ScanMap(Operator):
         defaults.det_data, help="Observation detdata key for the timestream data"
     )
 
+    det_data_units = Unit(u.K, help="Desired units of detector data")
```

Comment:
Ok, this impacts all such operators.  So your desired behavior would be:
- The det_data_units could still be specified for operators that generate timestreams
- If data exists, then the `ensure()` method should ignore any specified units (rather than raise)
- The operator should then get the units of the detector data (which might be different than the trait value) and do any conversions needed when accumulating to that field.
Will work on this.

Reply:
The ensure method now takes a `create_units` parameter which is only used when creating or remapping data objects.  There is also a new simple utility function to get the scale factor from source to target units.